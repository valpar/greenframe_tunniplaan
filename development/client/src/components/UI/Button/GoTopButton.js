import { useEffect, useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleUp } from "@fortawesome/free-regular-svg-icons";

const GoTopButton = () => {
  const [showGoTop, setShowGoTop] = useState(false);

  const handleVisibleButton = () => {
    setShowGoTop(window.pageYOffset > 50);
  };

  const handleScrollUp = () => {
    window.scrollTo({ left: 0, top: 0, behavior: "smooth" });
  };

  useEffect(() => {
    window.addEventListener("scroll", handleVisibleButton);

    return () => window.removeEventListener("scroll", handleVisibleButton);
  }, []);

  return (
    <div
      className={
        showGoTop
          ? `sticky flex justify-center lg:justify-end bottom-4 md:bottom-1 z-10`
          : `hidden`
      }
      onClick={handleScrollUp}
    >
      <FontAwesomeIcon
        data-testid="go-top-button"
        className="block text-4xl md:text-6xl cursor-pointer text-darkGray opacity-70 hover:opacity-100 duration-200"
        icon={faCircleUp}
      />
    </div>
  );
};
// top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2
export default GoTopButton;
