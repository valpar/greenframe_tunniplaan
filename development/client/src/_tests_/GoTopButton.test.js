import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import GoTopButton from '../components/UI/Button/GoTopButton.js';


describe('GoTopButton', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<GoTopButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render and handle scroll to top', () => {
    // Mock the scroll function
    const scrollToMock = jest.fn();
    window.scrollTo = scrollToMock;

    // Mock the scrollY value to simulate scrolling
    Object.defineProperty(window, 'scrollY', { value: 100 });

    // Render the component
    render(<GoTopButton />);

    // Check if a component value exists
    const goTopButton = screen.getByTestId('go-top-button');
    const ariaHiddenValue = goTopButton.getAttribute('aria-hidden');
    expect(ariaHiddenValue).toBe('true');

    // Simulate a scroll event to hide the button
    Object.defineProperty(window, 'scrollY', { value: 0 });
    fireEvent.scroll(window);
    //expect(ariaHiddenValue).toBe('false');

    // After scrolling, the button should be visible
    expect(screen.getByTestId('go-top-button')).toBeInTheDocument();
    
  });
});
