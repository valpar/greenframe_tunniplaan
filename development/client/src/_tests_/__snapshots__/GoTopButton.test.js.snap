// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoTopButton matches snapshot 1`] = `
<div
  className="hidden"
  onClick={[Function]}
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-circle-up block text-4xl md:text-6xl cursor-pointer text-darkGray opacity-70 hover:opacity-100 duration-200"
    data-icon="circle-up"
    data-prefix="far"
    data-testid="go-top-button"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM135.1 217.4c-4.5 4.2-7.1 10.1-7.1 16.3c0 12.3 10 22.3 22.3 22.3H208v96c0 17.7 14.3 32 32 32h32c17.7 0 32-14.3 32-32V256h57.7c12.3 0 22.3-10 22.3-22.3c0-6.2-2.6-12.1-7.1-16.3L269.8 117.5c-3.8-3.5-8.7-5.5-13.8-5.5s-10.1 2-13.8 5.5L135.1 217.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
